### PR TITLE
Fixes unsafe reducer-creators.

### PR DESCRIPTION
**Pull request by flamingbear**
Fixes CUMULUS-1988

Updates the reducer-creators to use safe methods not assuming that all values exist.

Adds tests for a small number of reducers as well as for the reducer-creators.

### CODE REVIEW

File: app/src/js/reducers/providers.js [hunk 1]
```diff
@@ -102,12 +102,6 @@ export default createReducer(initialState, {
     state.list.inflight = false;
     state.list.error = false;
   },
-  [PROVIDERS]: (state, action) => {
```
Comment (laurenfrederick) [inline app/src/js/reducers/providers.js:105]: Just because I don't know things... why is this removed?

Comment (flamingbear) [reply app/src/js/reducers/providers.js:105]: because it was duplicated in the code.
